[AMB-2505] feat: add Price Oracle configuration (#14)
apotdevin/thunderhub-docs: content/litd/price-oracle.mdx

import { Callout } from 'nextra/components';

# Price Oracle Configuration

> When trading [Taproot Assets](https://docs.lightning.engineering/the-lightning-network/taproot-assets) (such as [USDTL](https://www.speed.app/usdtl/)) through ThunderHub or [RailsX](https://amboss.tech/railsx), your litd node performs **sanity checks** on asset pricing during channel negotiations. These checks rely on a **[price oracle](https://www.cube.exchange/what-is/price-oracle)** — an external service that provides exchange rate information for Taproot Assets.

Reviewer: I wouldn't use the term "sanity checks" in formal documentation.

I wouldn't link to cube.exchange. They write about Ethereum and crypto stuff, could be off-putting for our audience.

```suggestion
When trading [Taproot Assets](https://docs.lightning.engineering/the-lightning-network/taproot-assets) (such as [USDTL](https://www.speed.app/usdtl/)) through ThunderHub and [RailsX](https://amboss.tech/railsx), your litd node performs **safety checks** on returned asset prices during asset swap negotiations to see whether the quoted prices are within an acceptable margin. These checks rely on a price oracle — an external service that provides exchange rate information for Taproot Assets.
```

Author: > I wouldn't link to cube.exchange. They write about Ethereum and crypto stuff, could be off-putting for our audience.

Yeah makes sense, I was doubtful about putting it too.

Your node must be configured with a price oracle that knows about the assets you want to trade. Without this, the node will reject trades because it cannot verify that the pricing is reasonable.

## Amboss Price Oracle (Recommended)

> The [Amboss](https://amboss.tech) price oracle knows about all assets available for trading in ThunderHub and RailsX, including USDTL. Add the following to your `lit.conf`:

Reviewer: ```suggestion
The [Amboss](https://amboss.tech) price oracle knows about all assets available for trading in ThunderHub and RailsX. Add the following to your `lit.conf`:
```

> **Mainnet:**

Reviewer: I think an explanation on how you can do this via Voltage would also be good.

Author: Added in both Voltage and the PO guide.

<img width="1291" height="332" alt="image" src="https://github.com/user-attachments/assets/4bfcbf07-3a56-48b1-ba86-710443b1a70b" />
<hr>
<img width="1248" height="554" alt="image" src="https://github.com/user-attachments/assets/809fbc65-6694-4c89-b297-159465f30326" />

```ini copy
taproot-assets.experimental.rfq.priceoracleaddress=rfqrpc://price-oracle.amboss.tech:443
```

**Testnet / Mutinynet:**

```ini copy
taproot-assets.experimental.rfq.priceoracleaddress=rfqrpc://price-oracle-dev.amboss.tech:443
```

<Callout type="info">
  The Amboss price oracle is kept in sync with all assets supported for trading
  on ThunderHub and RailsX. This is the simplest path to get started.
</Callout>

> ## Disable Sanity Checks

Reviewer: ```suggestion
## Disable Price Checks
```

> If you prefer not to use a price oracle, you can disable the pricing sanity checks entirely. Add the following to your `lit.conf`:

Reviewer: ```suggestion
If you prefer to not use a price oracle, you can instead opt to disable the price quote safety checks. This does however mean that your node will automatically accept whatever swap price the counterparty node returns.

 To do so, add the following to your `lit.conf`:
```

```ini copy
taproot-assets.experimental.rfq.skipacceptquotepricecheck=true
```

<Callout type="warning">

>   Disabling sanity checks means your node will accept trades without verifying

Reviewer: ```suggestion
  Disabling safety margin checks means your node will accept trades without verifying
```
